test/libsolidity/SolidityOptimizer.cpp

```diff
@@ -708,6 +708,20 @@ BOOST_AUTO_TEST_CASE(shift_optimizer_bug)
 	compareVersions("g(uint256)", u256(-1));
 }
 
+BOOST_AUTO_TEST_CASE(avoid_double_cleanup)
+{
+	char const* sourceCode = R"(
+		contract C {
+			receive() external payable {
+				abi.encodePacked(uint200(0));
+			}
+		}
+	)";
+	compileBothVersions(sourceCode, 0, "C", 50);
+	// This is counting in the deployed code.
+	BOOST_CHECK_EQUAL(numInstructions(m_nonOptimizedBytecode, Instruction::AND), 1);
+	BOOST_CHECK_EQUAL(numInstructions(m_optimizedBytecode, Instruction::AND), 0);
```

Comment:
This line is not needed and might be misleading into thinking that the point of the test is to check that the optimizer removes the and. Can you add a comment that we want the non-optimized code to only contain a single and?